app/src/main/java/com/example/development/sakaiclientandroid/fragments/AllCoursesFragment.java

```diff
@@ -160,29 +159,19 @@ public void createTreeView(ArrayList<ArrayList<Course>> coursesSorted)
 
                 courseNode.setClickListener(new TreeNode.TreeNodeClickListener() {
 
-                    //when click a course Node, open the CourseFragment to show
+                    //when click a course Node, open the CourseSitesFragment to show
                     //course specific information
                     @Override
                     public void onClick(TreeNode node, Object value)
                     {
                         if(value instanceof CourseHeaderViewHolder.CourseHeaderItem) {
-                            String courseSiteId = ((CourseHeaderViewHolder.CourseHeaderItem) value).siteId;
-
-                            Bundle bun = new Bundle();
-                            bun.putString(getString(R.string.site_id), courseSiteId);
-                            CourseFragment fragment = new CourseFragment();
-                            fragment.setArguments(bun);
-
 
                             //here we should save tree state
                             SharedPrefsUtil.saveTreeState(mContext, treeView, SharedPrefsUtil.ALL_COURSES_TREE_TYPE);
 
-                            //replaces current Fragment with CourseFragment
-                            FragmentTransaction ft = getActivity().getSupportFragmentManager().beginTransaction();
-                            ft.setCustomAnimations(R.anim.enter, R.anim.exit, R.anim.pop_enter, R.anim.pop_exit)
-                                    .replace(R.id.fragment_container, fragment, null)
-                                    .addToBackStack(null)
-                                    .commit();
+                            String courseSiteId = ((CourseHeaderViewHolder.CourseHeaderItem) value).siteId;
```

Comment:
@akench  Consider adding some null or class cast exception checks here since the nesting is two levels deep.